heat/core/statistics.py

```diff
@@ -376,6 +377,72 @@ def average(x, axis=None, weights=None, returned=False):
     return result
 
 
+def bincount(x, weights=None, minlength: int = 0):
+    """
+    Count number of occurrences of each value in array of non-negative ints.
+
+    The number of bins (size 1) is one larger than the largest value in `x`
+    unless `x` is empty, in which case the result is a tensor of size 0.
+    If `minlength` is specified, the number of bins is at least `minlength` and
+    if `x` is empty, then the result is tensor of size `minlength` filled with zeros.
+    If `n` is the value at position `i`, `out[n] += weights[i]` if weights is specified else `out[n] += 1`.
+
+    Parameters
+    ----------
+    x : DNDarray, 1 dimensional, non-negative ints
+    weights : DNDarray, optional
+        Weight for each value in the input tensor. Array of the same shape as x. Same split as `x`.
+    minlength : int, non-negative, optional
+        Minimum number of bins
+
+    Returns
+    -------
+    out : DNDArray
+        An array of length `max(x) + 1` if input is non-empty, else 0.
+
+    Examples
+    --------
+    >>> ht.bincount(ht.arange(5))
+    DNDarray([1, 1, 1, 1, 1], dtype=ht.int64, device=cpu:0, split=None)
+    >>> ht.bincount(ht.array([0, 1, 3, 2, 1]), weights=ht.array([0, 0.5, 1, 1.5, 2]))
+    DNDarray([0.0000, 2.5000, 1.5000, 1.0000], dtype=ht.float32, device=cpu:0, split=None)
+
+    Raises
+    ------
+    ValueError
+        If split axes of `x` and `weights` are not the same.
```

Comment:
It happens in the one case with None and 0, but pytorch would be complaining anyway because of different input tensor lengths.

Reply:
Yes, I understood what you were referring to, I was just suggesting to reword this line because `None` is not a split axis.  Suggestions:

- "if the `split` attribute is not the same"?  
- "if `x` is distributed and `weights` isn't, or vice versa"?